datafusion/substrait/src/logical_plan/consumer.rs

```diff
@@ -21,23 +21,24 @@ use datafusion::arrow::array::{GenericListArray, MapArray};
 use datafusion::arrow::datatypes::{
     DataType, Field, FieldRef, Fields, IntervalUnit, Schema, TimeUnit,
 };
-use datafusion::common::plan_err;
 use datafusion::common::{
-    not_impl_err, plan_datafusion_err, substrait_datafusion_err, substrait_err, DFSchema,
-    DFSchemaRef,
+    not_impl_datafusion_err, not_impl_err, plan_datafusion_err, plan_err,
+    substrait_datafusion_err, substrait_err, DFSchema, DFSchemaRef,
 };
 use datafusion::datasource::provider_as_source;
 use datafusion::logical_expr::expr::{Exists, InSubquery, Sort};
 
 use datafusion::logical_expr::{
-    Aggregate, BinaryExpr, Case, EmptyRelation, Expr, ExprSchemable, LogicalPlan,
-    Operator, Projection, SortExpr, TryCast, Values,
+    Aggregate, BinaryExpr, Case, Cast, EmptyRelation, Expr, ExprSchemable, Extension,
+    LogicalPlan, Operator, Projection, SortExpr, Subquery, TryCast, Values,
 };
 use substrait::proto::aggregate_rel::Grouping;
+use substrait::proto::expression as substrait_expression;
 use substrait::proto::expression::subquery::set_predicate::PredicateOp;
 use substrait::proto::expression_reference::ExprType;
 use url::Url;
 
+use super::state::SubstraitPlanningState;
```

Comment:
here as well, is this actually used? I don't see any references to it anymore

Reply:
`SubstraitPlanningState` is used in `producer.rs` still, so we can't remove it yet.

The import here brings the `table()` fn here https://github.com/apache/datafusion/blob/82a40f3edf264834c28f3a1bf2a3143951802634/datafusion/substrait/src/logical_plan/state.rs#L54-L62
into scope. It could make sense to inline it to remove it.

Reply:
Ah I see, yep I'd be for inlining it here, but not a critical thing then!

Reply:
Went ahead and inlined it to be able to remove all usages from the consumer. Will make it easier to remove fully once we updated the producer as well.